### DISCUSSION TITLE
Implementation of Google recaptcha for auth/login api

### DISCUSSION THREAD

**Discussion by workatease** [directus/directus#4079] [Archive]
Hi Team... I am trying to implement  google recaptcha as an extension for on auth/login before hook. to validate the service.
I see that the action is not supported and it it [API-hooks docs](https://docs.directus.io/guides/api-hooks/#event-format-options )TBD 
![image](https://user-images.githubusercontent.com/57198612/108009418-a8c07d00-6fd0-11eb-8126-fdebb74f8dd1.png)
 any idea on how to implement it.

**Comment by rijkvanzanten** [comment]
As you correctly stated, the auth hooks aren't implemented yet. Feel free to open a PR for those though, as they are something we're aiming to have included 🙂

**Reply by philleepflorence** [reply]
Is this the recommended starting point before contributing: [Contributing Docs](https://docs.directus.io/contributing/introduction/)

We also need to update the link in [directus/contributing.md](https://github.com/directus/directus/blob/main/contributing.md)

**Reply by workatease** [reply]
@rijkvanzanten I am working on implementing the auth api hooks, if we could discuss on what is the best option and place to implement it

**Reply by rijkvanzanten** [reply]
I think in here: https://github.com/directus/directus/blob/main/api/src/services/authentication.ts#L38 in a similar way to:

https://github.com/directus/directus/blob/2a738b7b52cafb1105cc7c51f3caaa312958e8f8/api/src/services/items.ts#L61-L69

**Reply by workatease** [reply]
so the scope will be **auth** and actions will be **login.before** 

auth.login.success
auth.login.fail
what all action can I add to the list

**Reply by rijkvanzanten** [reply]
Yeah!

I think it'd be

```
auth.login.before (blocking, allows for Captcha check / other things)
auth.login (generic on login, non-awaited when login is done)
```

I think instead of having 2 events for success/failure, we could have a single event and pass the state in the parameter of the hook

**Reply by workatease** [reply]
#4255 Submitted a Draft PR. Please check if this is good

**Comment by rijkvanzanten** [comment]
This is being implemented in https://github.com/directus/directus/pull/4255/
